### ISSUE TITLE
Copy Image from Coral Dev Board to Another

### ISSUE THREAD

**Issue by deepanalyticszlc** [google-coral/edgetpu#751]
I have been attempting to use the following protocol to copy an image from one coral dev board to another. I need to make 30 of the same system and while I have used the procedure successfully in the past, it no longer seems to work. Is their a preferred method to accomplish this task? If not, could the model of board be playing a role? The image I'd like to copy is on a board with #9316A00577. The new board is # 9423A00108. Any help is greatly appreciated.

# Back-Up/Copy Coral Dev Board Image 

## References:

https://coral.ai/docs/dev-board/serial-console/#connect-with-linux
https://stackoverflow.com/questions/64176284/backing-up-and-restoring-the-emmc

### How to copy image from coral dev board to PC
1. Board dip switches should be **on-off-off-off**
2. connect usb-b to serial and usb-c to OTG - ***Don't power yet***
3. In PC Terminal, not Coral:
    ```
    sudo usermod -aG plugdev,dialout <username> #i.e. zacharyclements
    ```
4. Restart computer
5. In PC Terminal:
    ```
    dmesg | grep ttyUSB
    ```
    You should see the output below:
    ```
    [  255.324887] usb 1-1: cp210x converter now attached to ttyUSB0
    [  255.326674] usb 1-1: cp210x converter now attached to ttyUSB1
    ```
6. Using the name from the first line, start a screen session with the following (you may need to `sudo apt install screen`):
    ```
    screen /dev/ttyUSB0 115200 #in screen
    ```
7. The first time you will need to press `enter` and enter a UN and PW, which by default are `mendel` for each:
8. Once screen session started, plugin power and hit `any key` within 2sec to enter U-Boot
9. In Screen Terminal:
    ```
    ums 0 mmc 0 
    ```
    This will mount the coral as (3) USB devices

10. In PC Terminal:
    ```
    fdisk -l 
    ```
    Locate the 8gig coral device and copy the path name. Should look like this, you want the disk name:
    ```
    Disk /dev/sdb: 7.29 GiB, 7818182656 bytes, 15269888 sectors
    Disk model: UMS disk 0      
    Units: sectors of 1 * 512 = 512 bytes
    Sector size (logical/physical): 512 bytes / 512 bytes
    I/O size (minimum/optimal): 512 bytes / 512 bytes
    Disklabel type: gpt
    Disk identifier: 85B83F23-51C4-4403-8BAC-906FF5800F4E

    Device       Start      End  Sectors  Size Type
    /dev/sdb1    16384   278527   262144  128M unknown
    /dev/sdb2   278528   280575     2048    1M unknown
    /dev/sdb3   294912  4489215  4194304    2G unknown
    /dev/sdb4  4489216 15253503 10764288  5.1G unknown
    ```
11. In PC Terminal:

    *Note: **/dev/sdb** is path of coral dsk from step 10.  Make sure this path is right for you***
    ```
    sudo dd if=/dev/sdb of=./backup.img bs=4M status=progress
    ```
    This will copy the image to your current directory and label it ***backup.img***
12. In Screen Terminal: Press `ctrl+a k` to close window.
13. Unmount USB devices and power down coral

### How to load backup image from PC to new coral dev board
14. **With new Coral** repeat steps **1-2** & **5-10**
15. In PC Terminal:

    *Note: **/dev/sdb** is path of coral dsk from step 10. Make sure this path is right for you. Entering the path of another disk will rewrite that disk permanently*
    ```
    sudo dd if=./backup.img of=/dev/sdb bs=4M status=progress
    ```
16. Kill the `Screen` window and power down the board. Disconnect USB-B serial cable and you should be able to boot up.
17. **TO ADD:** IP, Hostname, etc. will be copied from original board. ***Enter process to change these.***

**Comment by hjonnala** [comment]
Hi, any idea at what step the process is failing and the reason? Have you tried the same process with a dev board using the default image? 


You can create a bootable sd card image from [mendel linux source](https://coral.googlesource.com/docs/+/refs/heads/master/GettingStarted.md), using `m docker-sdcard` command. 

But, Please check with [NXP community](https://community.nxp.com/t5/i-MX-Processors-Knowledge-Base/i-MX8-Boot-process-and-creating-a-bootable-image/ta-p/1101253), if there is anyway to create a bootable sd card image using dd command from existing imx8m board. Thanks!!

**Comment by georgeplourde** [comment]
Are you flashing to a brand new Coral SoM?
After you perform the dd command to flash the new image,  is the bootcmd environment variable in uboot being updated to:
bootcmd=ext2load mmc 0:1 ${loadaddr} boot.scr; source; boota mmc0 boot_a

I've used the dd command above to clone a Coral SoM and then flash that image to another SoM.   In cases for a brand new CoralSoM when it would boot after flashing it would stop in the boot loader with a statement of "Welcome to your new Coral EdgeTPU Development Board".
Once the bootcmd envrionment variable was added to the boot loader, it booted and ran as expected with the flashed image
